### ISSUE TITLE
[3.3.5] [Sunwell Plateau] Bosses Kalecgos and Sathrovarr the Corruptor

### ISSUE THREAD

**Issue by peycho** [TrinityCore/TrinityCore#24512]
**Description:**

The boss is generally active, but there is some problem with spawning portals. I can't validate the other phase fight (in the portals)

**Current behaviour:**

Portals are not spawned and you can't fight in the opposite dimension with Sathrovarr the Corruptor.
The boss can't be killed because you need to do the same in the other dimension.

![WoWScrnShot_042520_235641](https://user-images.githubusercontent.com/170853/80512493-3bcaf080-8986-11ea-9f45-44b05d5fbdaf.jpg)

When the HP of the boss goes too low (~1000-5000) the boss stop doing anything. After a white the boss is reset and everything's going to the same place as it was before.

**Expected behaviour:**
Kalecgos is the first boss in the Sunwell Plateau raid. This highly technical fight takes place in two different realms at the same time. In the normal realm the players face Kalecgos the dragon, while in the spectral realm Sathrovarr the Corruptor and Kalec, Kalecgos' half-elf form, fight for control over the dragon. The battle is won when Sathrovarr and the dragon are both subdued, and it is lost when Sathrovarr kills Kalec. Roughly every 30 seconds, an opportunity presents itself for players to change realm. A solid rotation has to be worked out on who switches when to the spectral realm. 

**Kalecgos**
Tauntable
Melee: ~5k per hit, never crushing
[Arcane Buffet](https://wotlk.evowow.com/?spell=45018): instant cast, 8 sec CD, 463-537 Arcane AoE plus stacking debuff increasing Arcane damage taken by 500 for 40 seconds, hits all players in the normal realm. Not dispellable, removable by magical immunity effects like [Divine Shield](https://wotlk.evowow.com/?spell=642), [Ice Block](https://wotlk.evowow.com/?spell=45438), and [Cloak of Shadows](https://wotlk.evowow.com/?spell=31224) only. Automatically removed when entering the spectral realm
[Frost Breath](https://wotlk.evowow.com/?spell=44799): 1 sec cast, 30 yard frontal cone frost AoE plus 12 second 75% attack speed debuff, dispellable
[Tail Lash](https://wotlk.evowow.com/?spell=45122): Rear area physical AoE for ~1.5k damage plus 2 sec stun
Wild Magic:: Random target random (de)buff, not dispellable:
[Wild Magic Green](https://wotlk.evowow.com/?spell=44978): Healing done increased by 100%.
[Wild Magic Purple](https://wotlk.evowow.com/?spell=45001): Casting time increased by 100%.
[Wild Magic Red](https://wotlk.evowow.com/?spell=45002): Chance to hit with melee and ranged attacks reduced by 50%.
[Wild Magic White](https://wotlk.evowow.com/?spell=45004): Damage done by critical hits increased by 100%.
[Wild Magic Blac](https://wotlk.evowow.com/?spell=45006)k: Increases threat generated by 100%.
[Wild Magic Blue](https://wotlk.evowow.com/?spell=45010): Spell and ability costs reduced by 50%.
[Spectral Blast](https://wotlk.evowow.com/?spell=44866): Every 15–25 seconds, Kalecgos targets a random raid member (never the current tank and nobody with the [Spectral Exhaustion](https://wotlk.evowow.com/?spell=44867) debuff) for about 2 seconds. After that time, 5k arcane AoE damage are dealt in an 8 yard radius, the raid member is teleported to the spectral realm, and for ~10 seconds, a portal is created which others can click to go there too
[Spectral Realm](https://wotlk.evowow.com/?search=Spectral+Realm): Players entering the spectral realm get this 60 second debuff, upon expiration they are ported back to the normal realm
[Spectral Exhaustion](https://wotlk.evowow.com/?spell=44867): After leaving the spectral realm, players receive this 60 second debuff which prevents them from entering the spectral realm
Enrage: When either Sathrovarr or Kalecgos get to 10% health, both enrage and substantially increase their damage output

**Sathrovarr the Corruptor**
Tauntable
Melee: 6-8k, can crush
[Corrupting Strike](https://wotlk.evowow.com/?spell=45029):: 9k shadow damage plus stun and DoT for 1k over 3 seconds, ~13 sec CD
[Curse of Boundless Agony](https://wotlk.evowow.com/?spell=45032): 30 second shadow DoT which deals increasing damage every second. When removed this Curse jumps to the next target (unlimited range). It also jumps if the affected player dies or the duration ends. The damage dealt doubles every 5 seconds, starting at 100:
0-5 sec = 100 dps
5-10 sec = 200 dps
10-15 sec = 400 dps
15-20 sec = 800 dps
20-25 sec = 1600 dps
25-30 sec = 3200 dps
[Shadow Bolt Volley](https://wotlk.evowow.com/?spell=45031): 5k shadow damage on a random target and up to two nearby others
Enrage: When either Sathrovarr or Kalecgos get to 10% health, both enrage and substantially increase their damage output

Some of the values maybe not 100% true.
Source: https://wow.gamepedia.com/Kalecgos_(tactics)

**Steps to reproduce the problem:**
Create raid group.
Enter Sunwell Plateau. start the fight with Kalecgos

**Branch(es):**
3.3.5

**TC rev. hash/commit:** 
400e2913179e0ff47518558f89b687161031967a

**Operating system:** Ubuntu 16.04 LTS

**Comment by Rushor** [comment]
maybe related somewhere in the script https://github.com/TrinityCore/TrinityCore/issues/24294 ?

**Comment by Voidx** [comment]
no changes?

**Comment by w5860363** [comment]
@Rushor Disabled spell scriptnames
then. cast 46019 46020
It can be directly transferred to the nearby. On the contrary, if the skills are working, they will be transferred to the boss anyway

And the Z axis is wrong

**Comment by saliven1970** [comment]
I also encountered this problem. I can't switch to the spectral realm, and I can't see Sathrovarr

**Comment by WowChenEd** [comment]
> I also encountered this problem. I can't switch to the spectral realm, and I can't see Sathrovarr. But I know that I used this exploration sunwell at ten years ago, which was normal at that time, and boss could be killed by GM skill 5(Touch of death), but now it can't.
> 
> I tested two situations: single brush and RAID mode. When single brush, when Kalecgos is hit below 20000HP, enter the banished mode, and declare failure when Sathrovarr controls the dragon. There will be no Spectral Blast and can't see Sathrovarr. In RAID mode, there will be no portal when we get to Spectral Blast, Kalecgos disappears, reappears after 60 seconds, and the original HP is restored.When the battle begins, enter. cast 44866（SPELL_SPECTRAL_BLAST_EFFECT）, you can see the AOE effect and a portal appears. Click the portal can goto the spectral realm, but there is no Sathrovarr in it.
------------------------------------
the same problem with me, how do you resolve it?

**Comment by DykeRz** [comment]
any fixes?

**Comment by everfresh-admin** [comment]
Sathrovarr and Human Kalecgos are in the dome below Dragon Kalecgos. The teleport spells are broken. Replacing them with a direct teleport works as a simple solution.

Replace: `DoCastSelf(SPELL_TELEPORT_BACK, true);`
With: `me->NearTeleportTo(1704.34, 928.17, 53.079, 0);`

Replace: `target->CastSpell(target, SPELL_SPECTRAL_REALM_TELEPORT, true);`
With: `target->NearTeleportTo(1704.34, 928.17, -74.558, 0);`

Replace: `target->CastSpell(target, SPELL_TELEPORT_BACK, true);`
With: `target->NearTeleportTo(1704.34, 928.17, 53.079, 0);`

**Comment by Aokromes** [comment]
that's not fix instead hack.

**Comment by everfresh-admin** [comment]
Yes totally, but sharing hackfixes could be useful for others running private servers who just want it to work. And sharing the findings could save diagnosing time for someone trying to do the real fix. I couldn't find any information like this for TC or AC and have been debugging this for several days on and off.

Just looks like 2 teleport spells are utterly broken. If the reason why is really mmaps like Voidx said in 2020, then I have no idea how to debug those generated files whatsoever and this is the best I can do. It's a deadend for me.

**Comment by klopez84** [comment]
Yes I am after the tier sets but this same thing happens to me. How can i get around this or has this been correct or are we going to see this corrected?
